test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { getContractAt } = ethers;
+const { expect } = require('chai');
+const { getRandomBytes32 } = require('./utils');
+const { deployContract } = require('../scripts/deploy');
+
+/**
+ * Helper function to extract an address from a bytes32 storage slot
+ * @param {string} slotValue - The bytes32 value from storage
+ * @returns {string} The EIP-55 checksummed address
```

Comment:
```suggestion
 * Convert bytes32 value to EIP-55 address
 * @param {string} bytes - The bytes32 value
 * @returns {string} The EIP-55 checksummed address
```
the function is more generic than handling slot specific values